splunk_base/stackstate_checks/splunk/telemetry/splunk_telemetry_base.py

```diff
@@ -24,73 +40,42 @@ def __init__(self, name, init_config, agentConfig, persistence_check_name, insta
-                                         'value')
-            if 'token_auth' in authentication:
-                token_auth = authentication["token_auth"]
-                if 'name' not in token_auth:
-                    raise CheckException('Splunk event/metric instance missing "authentication.token_auth.name" value')
-                if 'initial_token' not in token_auth:
-                    raise CheckException('Splunk event/metric instance missing '
-                                         '"authentication.token_auth.initial_token" value')
-                if 'audience' not in token_auth:
-                    raise CheckException('Splunk event/metric instance missing "authentication.token_auth.audience" '
-                                         'value')
-
-        current_time = self._current_time_seconds()
-        url = instance["url"]
-        if url not in self.instance_data:
-            self.instance_data[url] = self.get_instance(instance, current_time)
-
-        instance = self.instance_data[url]
-        if not instance.initial_time_done(current_time):
-            self.log.debug("Skipping splunk metric/event instance %s, waiting for initial time to expire" % url)
-            return
+
+        if self.instance_data is None:
+            self.instance_data = self._build_instance(instance)
+
+        committable_state = CommittableState(self.commit_state, self.load_state(instance))
+
+        instance = self.instance_data
+
+# snapshot needed or not? self.snapshot = bool(instance.get('snapshot', True))
 
         try:
-            if authentication and 'token_auth' in authentication:
-                self.log.debug("Using token based authentication mechanism")
-                base_url = instance.instance_config.base_url
-                self._token_auth_session(instance, authentication, base_url, self.status, self.persistence_check_name)
-            else:
-                self.log.debug("Using basic authentication mechanism")
-                self._auth_session(instance)
-
-            saved_searches = self._saved_searches(instance)
-            instance.saved_searches.update_searches(self.log, saved_searches)
-
-            # after updating the regex searches in the saved search list, load the state and check
-            self.load_status()
-            instance.update_status(current_time, self.status)
-
-            executed_searches = False
-            for saved_searches in chunks(instance.saved_searches.searches, instance.saved_searches_parallel):
-                executed_searches |= self._dispatch_and_await_search(instance, saved_searches)
-
-            if len(instance.saved_searches.searches) != 0 and not executed_searches:
-                raise CheckException("No saved search was successfully executed.")
-
-            # If no service checks were produced, everything is ok
-            if len(self.service_checks) is 0:
-                self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.OK)
```

Comment:
This erased part should be covered in instance.saved_searches.run_saved_searches(_process_data, _service_check, self.log, committable_state), since in process_data  parameter is data from extract_telemetry()